internal/commands/tunnel/tunnel_up.go

```diff
@@ -287,22 +290,13 @@ type terraformOutput struct {
 }
 
 func sendSSHPublicKey(bastionID string, key string, sess *session.Session) error {
-	// This command is executed in the bastion host and it checks if our public key is present. If it's not it uploads it to _authorized_keys file.
-	command := fmt.Sprintf(
-		`grep -qR "%s" /home/ubuntu/.ssh/authorized_keys || echo "%s" >> /home/ubuntu/.ssh/authorized_keys`,
-		key, key,
-	)
-
-	_, err := ssm.New(sess).SendCommand(&ssm.SendCommandInput{
-		InstanceIds:  []*string{&bastionID},
-		DocumentName: aws.String("AWS-RunShellScript"),
-		Comment:      aws.String("Add an SSH public key to authorized_keys"),
-		Parameters: map[string][]*string{
-			"commands": {&command},
-		},
+	_, err := ec2instanceconnect.New(sess).SendSSHPublicKey(&ec2instanceconnect.SendSSHPublicKeyInput{
```

Comment:
👍🏻 Thanks for this.
Please confirm this is an idempotent function and every time we execute it it writes the key only once.